domain/storageprovisioning/state/state.go

```diff
@@ -288,3 +315,108 @@ func (st *State) checkApplicationExists(
 
 	return true, nil
 }
+
+// GetStorageIDsForUnit returns the storage IDs for the given unit UUID.
+//
+// The following errors may be returned:
+// - [applicationerrors.UnitNotFound] when no unit exists for the supplied unit UUID.
+// - [corestorage.InvalidStorageID] when the provided unit UUID is invalid.
+func (s *State) GetStorageIDsForUnit(ctx context.Context, unitUUID string) ([]corestorage.ID, error) {
+	db, err := s.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	input := storageAttachment{
```

Comment:
input should be of type `unitUUID` and remove the unit uuid off of `storageAttachment`. Because we don't want to use one more type that we have we are now dragging out n * unit uuid from the database for no reason. We should just pull the data we want.